swiftemulator/emulators/base.py

```diff
@@ -123,3 +123,90 @@ def predict_values(
             )
 
         raise NotImplementedError
+
+    def interactive_plot(
+        self,
+        x: np.array,
+        xlabel: str = "",
+        ylabel: str = "",
+        x_data: np.array = None,
+        y_data: np.array = None,
+    ):
+        """
+        Generates an interactive plot which displays the emulator predictions.
+        If no reference data is passed to be overplotted then the plot will
+        display a line which corresponds to the predictions for the mean
+        of the parameter values.
+
+        Parameters
+        ----------
+
+        x: np.array
+            Array of data for which the emulator should make predictions.
+
+        xlabel: str, optional
+            Label for horizontal axis on the resultant figure.
+
+        ylabel: str, optional
+            Label for vertical axis on the resultant figure.
+
+        x_data: np.array, optional
+            Array containing x-values of reference data to plot.
+
+        y_data: np.array, optional
+            Array containing y-values of reference data to plot.
+            Must be the same shape as x_data
+        """
+        import matplotlib.pyplot as plt
+        from matplotlib.widgets import Slider
+
+        fig, ax = plt.subplots()
+        model_specification = self.model_specification
+        param_means = {}
+        sliders = []
+        n_param = model_specification.number_of_parameters
+        fig.subplots_adjust(bottom=0.12 + n_param * 0.1)
+        for i in range(n_param):
+            # Extracting information needed for slider
+            name = model_specification.parameter_names[i]
+            lo_lim = sorted(model_specification.parameter_limits[i])[0]
+            hi_lim = sorted(model_specification.parameter_limits[i])[1]
+            param_means[name] = (lo_lim + hi_lim) / 2
```

Comment:
Maybe refer to it as midpoints as that is more correct (also clarifies my earlier comment)